### PR TITLE
Enhancements to Diagnostics

### PR DESCRIPTION
**Pull request by justokman**
- Stats initialized implicitly with deprecation warnings for calls to init(), initStats()
- Supplanting local stats parsing with @dolbyio/webrtc-stats library
- Diagnose.get() now accepts a config object with the addition of 'format' for CMCD compatible data
- Changing default stats array length to 60 for 1 minutes data

### CODE REVIEW

File: packages/millicast-sdk/src/PeerConnectionStats.js [hunk 1]
```diff
@@ -62,221 +63,62 @@ export const peerConnectionStatsEvents = {
...
+      })
+      this.collection.on('stats', (stats) => {
+        Diagnostics.addStats(stats)
+        this.emit(peerConnectionStatsEvents.stats, stats)
+      })
+      this.collection.start()
+    } catch (e) {
+      logger.error(e)
+    }
+    this.initialized = true
```
Comment (dubeyShivank) [inline packages/millicast-sdk/src/PeerConnectionStats.js:103]: should initialized be set before `this.collection.start()` ?

File: packages/millicast-sdk/src/PeerConnectionStats.js [hunk 2]
```diff
@@ -62,221 +63,62 @@ export const peerConnectionStatsEvents = {
...
   }
 
   /**
    * Parse incoming RTCPeerConnection stats.
+   * @deprecated since version 0.1.45 - will be removed in future releases.
    * @param {RTCStatsReport} rawStats - RTCPeerConnection stats.
-   * @returns {ConnectionStats} RTCPeerConnection stats parsed.
+   * @returns {null} Method deprecated and no longer returns meaningful data.
    */
   parseStats (rawStats) {
```
Comment (dubeyShivank) [inline packages/millicast-sdk/src/PeerConnectionStats.js:112]: For customers who use the SDK via CDN and not NPM, this might result in a breaking change. 

I like the warning, but we might have to keep this method around for a bit. 

@j12y @vincentsong what do you guys think?

**Comment by Santiago-Souto** [comment]
@justokman please make sure to bring `main` branch as #340 was merged right now, and could bring some conflicts

File: packages/millicast-sdk/src/PeerConnectionStats.js (hunk 2 above)
Comment (j12y) [reply packages/millicast-sdk/src/PeerConnectionStats.js:112]: I may need some clarification on scope to answer, because if this is only used in the results of `diagnose()` then that is fine, we want `diagnose()` to return webrtc-stats library results and keys.

If this is a change to both `diagnose()` and the results of `webRTCPeer.on('stats', ...)` then that is more than we want to change at this time. There would be dependencies on hosted player, etc. we would need to reconcile for that deep of a change.

in other words:
- if diagnose is all that is impacted, this is fine
- if all `initStats()` users are impacted, then that's a bigger deal we should revisit

Comment (justokman) [reply packages/millicast-sdk/src/PeerConnectionStats.js:112]: @dubeyShivank It looks like this method never returned an object like that. 

It is used to parse the raw stats in the original interval and was exposed publicly.

I think it will be safe to deprecate it.

File: packages/millicast-sdk/src/PeerConnectionStats.js (hunk 1 above)
Comment (justokman) [reply packages/millicast-sdk/src/PeerConnectionStats.js:103]: Moving to try block.

File: packages/millicast-sdk/src/PeerConnectionStats.js (hunk 2 above)
Comment (dubeyShivank) [reply packages/millicast-sdk/src/PeerConnectionStats.js:153]: Ok sweet - thanks for confirming!

File: packages/millicast-sdk/package-lock.json [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
   "requires": true,
   "packages": {
     "": {
       "name": "@millicast/sdk",
-      "version": "0.1.44-rc-1",
+      "version": "0.1.44",
       "license": "See in LICENSE file",
       "dependencies": {
+        "@dolbyio/webrtc-stats": "^0.4.0",
+        "@peermetrics/webrtc-stats": "^5.7.1",
```
Comment (dubeyShivank) [inline packages/millicast-sdk/package-lock.json:13]: Just wanting to be sure - do we need `peermetrics` as well?

File: packages/millicast-sdk/src/PeerConnection.js [hunk 4]
```diff
@@ -43,11 +43,15 @@ export default class PeerConnection extends EventEmitter {
   /**
    * Instance new RTCPeerConnection.
    * @param {RTCConfiguration} config - Peer configuration.
+   * @param {Boolean} [autoInitStats = true] - True to initialize statistics monitoring of the RTCPeerConnection accessed via Logger.get(), false to opt-out.
    */
-  async createRTCPeer (config = null) {
+  async createRTCPeer (config = { autoInitStats: true }) {
```
Comment (dubeyShivank) [inline packages/millicast-sdk/src/PeerConnection.js:48]: Probably worth checking - I wasn't seeing stats by default in `Logger.diagnose()` on the publishing side. maybe my setup was fudged, I dunno - but would be good to cross-check

File: packages/millicast-sdk/package-lock.json (hunk 3 above)
Comment (justokman) [reply packages/millicast-sdk/package-lock.json:13]: Huh, I thought I removed that.  Not sure how it got there.

File: packages/millicast-sdk/src/PeerConnection.js (hunk 4 above)
Comment (justokman) [reply packages/millicast-sdk/src/PeerConnection.js:48]: Logger.diagnose() works as expected for me after starting a stream in the publisher demo.  We can sync up at standup on this.

File: packages/millicast-sdk/src/utils/Diagnostics.js [hunk 5]
```diff
@@ -10,6 +11,36 @@ let _feedId = ''
...
+    }
+  }
+
+  diagnostics.stats = diagnostics.stats.reduce((acc, stat) => {
+    const audioStats = stat.audio.inbounds.length !== 0
+      ? stat.audio.inbounds.map(statAudio => mapStats('audio', statAudio))
+      : stat.audio.outbounds.map(statAudio => mapStats('audio', statAudio))
+    const videoStats = stat.video.inbounds.length !== 0
+      ? stat.video.inbounds.map(statVideo => mapStats('video', statVideo))
+      : stat.video.outbounds.map(statVideo => mapStats('video', statVideo))
```
Comment (Santiago-Souto) [inline packages/millicast-sdk/src/utils/Diagnostics.js:36]: @dubeyShivank In refactored this lines as they were only considering inbound stats, so when testing publish side CMCD stats where not shown. Please review it as well.

Comment (dubeyShivank) [reply packages/millicast-sdk/src/utils/Diagnostics.js:36]: Thanks Santi, good call out. I will address it

**Review by dubeyShivank — DISMISSED** [review]

**Comment by Santiago-Souto** [comment]
Forced push commits to ensure signing. Will move on to merging this PR

**Review by Santiago-Souto — APPROVED** [review]
